### ISSUE TITLE
Error pop from empty list from telebot.__init__.py _notify_next_handlers

### ISSUE THREAD

**Issue by Saigesp** [eternnoir/pyTelegramBotAPI#548]
Sys
```
pyTelegramBotAPI version 3.6.4
Windows 10 Pro
Python 2.7.15
```

Error:
```
Traceback (most recent call last):
  File "build\bdist.win32\egg\telebot\util.py", line 59, in run
    task(*args, **kwargs)
  File "build\bdist.win32\egg\telebot\__init__.py", line 278, in __retrieve_updates
    self.process_new_updates(updates)
  File "build\bdist.win32\egg\telebot\__init__.py", line 315, in process_new_updates
    self.process_new_messages(new_messages)
  File "build\bdist.win32\egg\telebot\__init__.py", line 334, in process_new_messages
    self._notify_next_handlers(new_messages)
  File "build\bdist.win32\egg\telebot\__init__.py", line 1298, in _notify_next_handlers
    new_messages.pop(i)  # removing message that detects with next_step_handler
IndexError: pop from empty list
```

Error function:
```
    def _notify_next_handlers(self, new_messages):
        i = 0
        while i < len(new_messages):
            message = new_messages[i]
            chat_id = message.chat.id
            was_poped = False
            if chat_id in self.next_step_handlers.keys():
                handlers = copy.deepcopy(self.next_step_handlers[chat_id])
                self.next_step_handlers.pop(chat_id, None)
                if (handlers):
                    for handler in handlers:
                        self._exec_task(handler["callback"], message, *handler["args"], **handler["kwargs"])
                    new_messages.pop(i)  # removing message that detects with next_step_handler
                    was_poped = True
                new_messages.pop(i)  # removing message that detects with next_step_handler
                if self.next_step_saver is not None:
                    self.next_step_saver.start_save_timer()
            if (not was_poped):
                i += 1
```

As long as I can see, second pop() gives an error. It seems a duplicated line

**Comment by Saigesp** [comment]
Sorry, second line with comment:
```
new_messages.pop(i)  # removing message that detects with next_step_handler
```

**Comment by almaz5000** [comment]
same error

**Comment by eternnoir** [comment]
My mistake to merge pr with conflict. Fix at 3.6.5
